tensorflow-core/tensorflow-core-api/src/test/java/org/tensorflow/op/core/ShapeOpsTest.java

```diff
@@ -0,0 +1,305 @@
+            }
+            
+            size = instance.size(tfshape, Constant.scalarOf(scope, 2));
+            try (Tensor<TInt32> result = session.runner().fetch(size.asOutput()).run().get(0).expect(TInt32.DTYPE)) {
+                result.data().scalars().forEach(s -> assertEquals(1, s.getInt()));
+            }
+        }
+    }
+
+    /**
+     * Test of size method, of class ShapeOps.
+     */
+    @Test
+    public void testSize_Operand_Operand() {
+        System.out.println("size");
+         try (Graph g = new Graph();
+                Session session = new Session(g)) {
+            Scope scope = new Scope(g);
+            ShapeOps instance = ShapeOps.create(scope);
+            Operand operand = Constant.arrayOf(scope, new float[] {1, 2, 3, 4, 5, 6, 7, 8} );
+            Operand actual = Reshape.create(scope, operand, Constant.vectorOf(scope, new long[] { 4,2,1} ));
+            
+            Operand<TInt32> size = instance.size(actual, Constant.scalarOf(scope, 0));
+            try (Tensor<TInt32> result = session.runner().fetch(size.asOutput()).run().get(0).expect(TInt32.DTYPE)) {
+                result.data().scalars().forEach(s -> assertEquals(4, s.getInt()));
+            }
+            
+            size = instance.size(actual, Constant.scalarOf(scope, 1));
+            try (Tensor<TInt32> result = session.runner().fetch(size.asOutput()).run().get(0).expect(TInt32.DTYPE)) {
+                result.data().scalars().forEach(s -> assertEquals(2, s.getInt()));
+            }
+            
+            size = instance.size(actual, Constant.scalarOf(scope, 2));
+            try (Tensor<TInt32> result = session.runner().fetch(size.asOutput()).run().get(0).expect(TInt32.DTYPE)) {
+                result.data().scalars().forEach(s -> assertEquals(1, s.getInt()));
+            }
+        }
+    }
+
+    /**
+     * Test of numDimensions method, of class ShapeOps.
+     */
+    @Test
+    public void testNumDimensions() {
+        System.out.println("numDimensions");
+        try (Graph g = new Graph();
+                Session session = new Session(g)) {
+            Scope scope = new Scope(g);
+            ShapeOps instance = ShapeOps.create(scope);
+            Operand operand = Constant.arrayOf(scope, new float[] {1, 2, 3, 4, 5, 6, 7, 8} );
+            Operand actual = Reshape.create(scope, operand, Constant.vectorOf(scope, new long[] { 4,2,1} ));
+            Shape<TInt32> tfshape = Shape.create(scope, actual);
+            
+            Operand<TInt32> nDims = instance.numDimensions(tfshape);
+            try (Tensor<TInt32> result = session.runner().fetch(nDims.asOutput()).run().get(0).expect(TInt32.DTYPE)) {
+                result.data().scalars().forEach(s -> assertEquals(3, s.getInt()));
+            }
+            
+        }
+    }
+
+    /**
+     * Test of reduceDims method, of class ShapeOps.
+     */
+    @Test
+    public void testReduceDims_Operand_Operand() {
+        System.out.println("reduceDims");
```

Comment:
What's this test trying to do?

Reply:
My mistake, It was intended to be the same as testReduceDims_Shape_Operand, but based on passing the operand rather than the shape directly. It may be a redundant, unless you think we need to test it both ways.